File: MiniDOMFinanziaria/Finanziaria/RichiesteFinanziamenti/Azioni/PrendiInCaricoRichiestaFinanziamento.cs
using System;
using DMD;

namespace minidom
{
    public partial class Finanziaria
    {
        public class PrendiInCaricoRichiestaFinanziamento : Sistema.AzioneEseguibile
        {
            public override string Description
            {
                get
                {
                    return "Prende in carico la richiesta di finanziamento e programma un ricontatto nel CRM";
                }
            }

            protected override string ExecuteInternal(Sistema.Notifica notifica, string parameters)
            {
                if (notifica.SourceName != "CRichiestaFinanziamento")
                    throw new ArgumentException("L'azione non è definita sul tipo [" + notifica.SourceName + "]");
                var richFin = RichiesteFinanziamento.GetItemById(notifica.SourceID);
                if (richFin is null)
                    throw new ArgumentNullException("Richiesta di finaniamento");
                richFin.PrendiInCarico();
                var ric = Anagrafica.Ricontatti.ProgrammaRicontatto(richFin.Cliente, DMD.DateUtils.Now(), "Ricontatto programmato per la richiesta online: " + richFin.ToString(), DMD.RunTime.vbTypeName(richFin), DBUtils.GetID(richFin).ToString(), "", richFin.PuntoOperativo, Sistema.Users.CurrentUser);
                notifica.StatoNotifica = Sistema.StatoNotifica.LETTA;
                return DBUtils.GetID(ric).ToString();
            }

            public override string Name
            {
                get
                {
                    return "RICHFINPRENDIINCARICO";
                }
            }

            public override object Render(Sistema.Notifica notifica, object context)
            {
                return null;
            }
        }
    }
}

File: MiniDOMFinanziaria/Finanziaria/RichiesteFinanziamenti/Azioni/PrendiInCaricoRichiestaFinanziamento.vb
Imports minidom.Databases
Imports minidom.Anagrafica
Imports minidom.Sistema

Partial Public Class Finanziaria

    Public Class PrendiInCaricoRichiestaFinanziamento
        Inherits AzioneEseguibile


        Public Overrides ReadOnly Property Description As String
            Get
                Return "Prende in carico la richiesta di finanziamento e programma un ricontatto nel CRM"
            End Get
        End Property

        Protected Overrides Function ExecuteInternal(notifica As Notifica, parameters As String) As String
            If (notifica.SourceName <> "CRichiestaFinanziamento") Then Throw New ArgumentException("L'azione non è definita sul tipo [" & notifica.SourceName & "]")
            Dim richFin As CRichiestaFinanziamento = Finanziaria.RichiesteFinanziamento.GetItemById(notifica.SourceID)
            If (richFin Is Nothing) Then Throw New ArgumentNullException("Richiesta di finaniamento")
            richFin.PrendiInCarico()
            Dim ric As CRicontatto = Anagrafica.Ricontatti.ProgrammaRicontatto(richFin.Cliente, DateUtils.Now(), "Ricontatto programmato per la richiesta online: " & richFin.ToString, TypeName(richFin), GetID(richFin), "", richFin.PuntoOperativo, Users.CurrentUser)
            notifica.StatoNotifica = StatoNotifica.LETTA
            Return GetID(ric)
        End Function

        Public Overrides ReadOnly Property Name As String
            Get
                Return "RICHFINPRENDIINCARICO"
            End Get
        End Property

        Public Overrides Function Render(notifica As Notifica, context As Object) As Object
            Return Nothing
        End Function

    End Class

End Class